x-pack/plugins/observability_onboarding/public/hooks/use_system_integration_status.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import { useKibana } from './use_kibana';
+
+export const SYSTEM_INTEGRATION_URL = '/api/fleet/epm/packages/system';
+export const SYSTEM_INTEGRATION_STATUS_QUERY_KEY = ['system-integration'];
+
+type IntegrationInstallStatus =
+  | 'installed'
+  | 'installing'
+  | 'install_failed'
+  | 'not_installed';
+
+const INTEGRATIONS_CALL_TIMEOUT = 2000;
```

Comment:
Could this fail in some environments? Do we have to hardcode a number? And should we perhaps make it higher?